test/test_um2netcdf.py

```diff
@@ -0,0 +1,68 @@
+import unittest.mock as mock
+
+import umpost.um2netcdf as um2nc
+
+import pytest
+import mule
+
+
+def test_get_eg_grid_type():
+    ff = mule.ff.FieldsFile()
+    ff.fixed_length_header.grid_staggering = 6
+    grid_type = um2nc.get_grid_type(ff)
+    assert grid_type == um2nc.GRID_END_GAME
+
+
+def test_get_nd_grid_type():
+    ff = mule.ff.FieldsFile()
+    ff.fixed_length_header.grid_staggering = 3
+    grid_type = um2nc.get_grid_type(ff)
+    assert grid_type == um2nc.GRID_NEW_DYNAMICS
+
+
+def test_get_grid_type_error():
+    ff = mule.ff.FieldsFile()  # "empty" fields file has no grid staggering
+
+    with pytest.raises(um2nc.UMError):
```

Comment:
Can also use `match` to check the error message, e.g:
```
with pytest.raises(um2nc.UMError, match="Unable to determine grid staggering from header '{no_staggering_value}'"):
```

Reply:
That's a useful feature. I'll skip testing the error messages until usage requirements are clearer.